### DISCUSSION TITLE
Wordpiece tokenization directly in Java?

### DISCUSSION THREAD

**Discussion by lintool** [castorini/anserini#1822] [Ideas]
Apparently, there's already an implementation in djl: https://github.com/deepjavalibrary/djl/blob/master/api/src/main/java/ai/djl/modality/nlp/bert/WordpieceTokenizer.java

Javadoc:
https://javadoc.io/doc/ai.djl/api/latest/ai/djl/modality/nlp/bert/WordpieceTokenizer.html

**Comment by ToluClassics** [comment]
@lintool Apparently DJL has an implementation for loading Huggingface Tokenizers that works. Wondering if we can explore that instead?

https://github.com/deepjavalibrary/djl/tree/master/extensions/tokenizers

**Reply by lintool** [reply]
Yup, that's a promising lead.

**Reply by lintool** [reply]
We've done this a while ago... closing!
